yt_dlp/extractor/hse.py

```diff
@@ -0,0 +1,122 @@
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        json_str = self._html_search_regex(r'window\.__REDUX_DATA__ = ({.*});?', webpage, 'json_str')
+        json_str = json_str.replace('\n', '')
+        json_data = self._parse_json(json_str, video_id)
+
+        if not try_get(json_data, lambda x: x['tvShowPage']['tvShowVideo']['sources']):
+            raise ExtractorError('No show or video found', expected=True)
+
+        page = json_data['tvShowPage']
+        video = page['tvShowVideo']
+        formats = []
+        for src in video['sources']:
+            if src['mimetype'] == 'application/x-mpegURL':
+                formats.extend(self._extract_m3u8_formats(src['url'], video_id, ext='mp4'))
+        self._sort_formats(formats)
+        thumbnail = try_get(video, lambda x: x['poster'])
+
+        show = try_get(page, lambda x: x['tvShow'])
+        title = try_get(show, lambda x: x['title'])
+        uploader = try_get(show, lambda x: x['presenter'])
+        date = try_get(show, lambda x: x['date'])
+        hour = try_get(show, lambda x: x['hour'])
+        timestamp = unified_timestamp(f'{date} {hour}:00')
+
+        show_id = try_get(show, lambda x: x['actionFieldText'])
+        channel = self._search_regex(r'tvShow \| ([A-Z0-9]+)_', show_id, video_id)
+
+        return {
+            'id': video_id,
+            'title': title,
+            'formats': formats,
+            'timestamp': timestamp,
+            'thumbnail': thumbnail,
+            'channel': channel,
+            'uploader': uploader
+        }
+
+
+class HSEProductIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?hse\.de/dpl/p/product/(?P<id>[0-9]+)'
+    _GEO_COUNTRIES = ['DE']
+    _TESTS = [{
+        'url': 'https://www.hse.de/dpl/p/product/408630',
+        'info_dict': {
+            'id': '408630',
+            'ext': 'mp4',
+            'title': 'Hose im Ponte-Mix',
+            'uploader': 'Judith Williams'
+        },
+        'params': {
+            'skip_download': True,
+        },
+    },
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        json_str = self._html_search_regex(r'window\.__REDUX_DATA__ = ({.*});?', webpage, 'json_str')
+        json_str = json_str.replace('\n', '')
+        json_data = self._parse_json(json_str, video_id)
+
+        if not try_get(json_data, lambda x: x['productContent']['productContent']['videos']):
+            raise ExtractorError('No product or video found', expected=True)
+
+        formats = []
+        for video in json_data['productContent']['productContent']['videos']:
+            thumbnail = video['poster']
+            sources = video['sources']
+            for src in sources:
+                if src['mimetype'] == 'application/x-mpegURL':
+                    formats.extend(self._extract_m3u8_formats(src['url'], video_id, ext='mp4'))
```

Comment:
same as above...parse subs

Reply:
there are no subtitles...

Reply:
if they add them in future then the program will start throwing ugly warnings.
Therefore it's better to just handle subs now.

Reply:
ok